test/e2e/xsite/xsite_test.go

```diff
@@ -467,6 +468,87 @@ func TestSuspectAndHearbeatConfig(t *testing.T) {
 	expectHeartBeatConfiguration(t, testKubes["xsite2"], true, hbInterval, hbTimeout)
 }
 
+func TestMultipleSites(t *testing.T) {
+	// single cluster cross-site is enough.
+	testName := tutils.TestName(t)
+	clientConfig := clientcmd.GetConfigFromFileOrDie(kube.FindKubeConfig())
+	kube := tutils.NewTestKubernetes(clientConfig.CurrentContext)
+	namespace := fmt.Sprintf("%s-xsite2", tutils.Namespace)
+
+	defer kube.CleanNamespaceAndLogOnPanic(t, namespace)
+
+	infinispan := &ispnv1.Infinispan{
+		TypeMeta: tutils.InfinispanTypeMeta,
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   fmt.Sprintf("%s-site", strcase.ToKebab(testName)),
+			Labels: map[string]string{"test-name": testName},
+		},
+		Spec: ispnv1.InfinispanSpec{
+			Replicas: 3,
+			Service: ispnv1.InfinispanServiceSpec{
+				Type: ispnv1.ServiceTypeDataGrid,
+				Sites: &ispnv1.InfinispanSitesSpec{
+					Local: ispnv1.InfinispanSitesLocalSpec{
+						Name: "site",
+						Expose: ispnv1.CrossSiteExposeSpec{
+							Type: ispnv1.CrossSiteExposeTypeClusterIP,
+						},
+						MaxRelayNodes: 2,
+						Discovery: &ispnv1.DiscoverySiteSpec{
+							Memory: "500Mi",
+							CPU:    "500m",
+						},
+					},
+					Locations: []ispnv1.InfinispanSiteLocationSpec{
+						{
+							Name: "remote-site-1",
+							URL:  "infinispan+xsite://fake-site-1.svc.local:7900",
+						},
+						{
+							Name: "remote-site-2",
+							URL:  "infinispan+xsite://fake-site-2.svc.local:7900",
+						},
+						{
+							Name: "remote-site-3",
+							URL:  "infinispan+xsite://fake-site-3.svc.local:7900",
+						},
+						{
+							Name: "remote-site-4",
+							URL:  "infinispan+xsite://fake-site-4.svc.local:7900",
+						},
+					},
+				},
+				Container: &ispnv1.InfinispanServiceContainerSpec{
+					EphemeralStorage: true,
+				},
+			},
+			Container: ispnv1.InfinispanContainerSpec{
+				Memory: tutils.Memory,
+			},
+			ConfigListener: &ispnv1.ConfigListenerSpec{
+				// Disable the ConfigListener to reduce the total number of resources required
+				Enabled: false,
+			},
+		},
+	}
+
+	kube.CreateInfinispan(infinispan, namespace)
+	// the bug prevents the pod to be ready
+	kube.WaitForInfinispanPods(3, tutils.SinglePodTimeout, infinispan.Name, namespace)
+	kube.WaitForInfinispanCondition(infinispan.Name, namespace, ispnv1.ConditionWellFormed)
+	// cannot check for cross-site view since it won't be formed
+
+	version := kube.GetStatefulSet(infinispan.Name, infinispan.Namespace).ObjectMeta.ResourceVersion
+
+	time.Sleep(10 * time.Second)
+
+	kube.WaitForInfinispanPods(3, tutils.SinglePodTimeout, infinispan.Name, namespace)
+	kube.WaitForInfinispanCondition(infinispan.Name, namespace, ispnv1.ConditionWellFormed)
+
+	otherVersion := kube.GetStatefulSet(infinispan.Name, infinispan.Namespace).ObjectMeta.ResourceVersion
+	assert.Equal(t, version, otherVersion)
```

Comment:
@ryanemerson is there a better way to test it?

Reply:
I think you should check the StatefulSet `ObservedGeneration` is 1 as the StatefulSet should never be modified once the initial cluster has been created. Take a look at `spec_update_test` for an idea of how to retrieve and check this.

Reply:
I'll try the generation... it is failing randomly locally :cry: